cli/cmd/deploy/deploy.go

```diff
@@ -358,6 +388,169 @@ func loginWithTelemetry(ctx context.Context, ch *cmdutil.Helper, redirectURL str
+
+		if res.GrantAccessUrl != "" {
+			// Print instructions to grant access
+			time.Sleep(3 * time.Second)
+			ch.Print("Open this URL in your browser to grant Rill access to Github:\n\n")
+			ch.Print("\t" + res.GrantAccessUrl + "\n\n")
+
+			// Open browser if possible
+			_ = browser.Open(res.GrantAccessUrl)
+		}
+	}
+
+	// Poll for permission granted
+	pollCtx, cancel := context.WithTimeout(ctx, pollTimeout)
+	defer cancel()
+	for {
+		select {
+		case <-pollCtx.Done():
+			return pollCtx.Err()
+		case <-time.After(pollInterval):
+			// Ready to check again.
+		}
+
+		// Poll for access to the Github's user account
+		pollRes, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+		if err != nil {
+			return err
+		}
+		if !pollRes.HasAccess {
+			// Sleep and poll again
+			continue
+		}
+
+		// Emit success telemetry
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedSuccess)
+
+		repoOwner := pollRes.Account
+		if len(pollRes.Organizations) > 0 {
+			repoOwners := []string{pollRes.Account}
+			repoOwners = append(repoOwners, pollRes.Organizations...)
+			ch.Print("\nYou also have access to organization(s)\n\n")
+			repoOwner = cmdutil.SelectPrompt("Please chhose where to create the repository", repoOwners, pollRes.Account)
+		}
+		// create and verify
+		githubRepository, err := createGithubRepository(ctx, ch, pollRes, localGitPath, repoOwner)
+		if err != nil {
+			return err
+		}
+
+		// init git repo
+		repo, err := git.PlainInit(localGitPath, false)
+		if err != nil {
+			if !errors.Is(err, git.ErrRepositoryAlreadyExists) {
+				return fmt.Errorf("failed to init git repo: %w", err)
+			}
+			repo, err = git.PlainOpen(localGitPath)
+			if err != nil {
+				return fmt.Errorf("failed to open git repo: %w", err)
+			}
+		}
+
+		wt, err := repo.Worktree()
+		if err != nil {
+			return fmt.Errorf("failed to get worktree: %w", err)
+		}
+
+		// git add .
+		if err := wt.AddWithOptions(&git.AddOptions{All: true}); err != nil {
+			return fmt.Errorf("failed to add files to git: %w", err)
+		}
+
+		// git commit -m
+		_, err = wt.Commit("Auto committed by Rill", &git.CommitOptions{All: true})
+		if err != nil {
+			if !errors.Is(err, git.ErrEmptyCommit) {
+				return fmt.Errorf("failed to commit files to git: %w", err)
+			}
+		}
+
+		// Create the remote
+		_, err = repo.CreateRemote(&config.RemoteConfig{Name: "origin", URLs: []string{*githubRepository.HTMLURL}})
+		if err != nil {
+			return fmt.Errorf("failed to create remote: %w", err)
+		}
+
+		// push the changes
+		if err := repo.PushContext(ctx, &git.PushOptions{Auth: &http.BasicAuth{Username: "x-access-token", Password: pollRes.AccessToken}}); err != nil {
+			return fmt.Errorf("failed to push to remote %q : %w", *githubRepository.HTMLURL, err)
+		}
+
+		printer.ColorGreenBold.Printf("\nRepository %q created successfully. Local changes pushed to remote.\n\n", *githubRepository.Name)
```

Comment:
Would suggest printing repo created message right after it is created in case of errors during push. And then print something like "pushing local project to Github..." while doing the push.